tedana/gscontrol.py

```diff
@@ -135,88 +155,91 @@ def minimum_image_regression(optcom_ts, mmix, mask, comptable, io_generator):
+    IOGenerator Label                           Content
+    ========================================    =================================================
+    "t1 like img"                               T1-like effect
+    "mir denoised img"                          Denoised version of T1-corrected time series
+    "ICA MIR mixing tsv"                        T1 global signal-corrected mixing matrix
+
+    if io_generator.verbose==True
+    "ICA accepted mir denoised img"             T1-corrected BOLD (high-Kappa) time series
+    "ICA accepted mir component weights img"    T1 global signal-corrected components
+    ========================================    =================================================
 
     References
     ----------
     .. footbibliography::
     """
     LGR.info("Performing minimum image regression to remove spatially-diffuse noise")
     RepLGR.info(
-        "Minimum image regression was then applied to the "
-        "data in order to remove spatially diffuse noise \\citep{kundu2013integrated}."
+        "Minimum image regression was then applied to the data in order to remove spatially "
+        "diffuse noise \\citep{kundu2013integrated}."
     )
 
     all_comps = comptable.index.values
     acc = comptable[comptable.classification == "accepted"].index.values
     ign = comptable[comptable.classification == "ignored"].index.values
     not_ign = sorted(np.setdiff1d(all_comps, ign))
 
-    optcom_masked = optcom_ts[mask, :]
-    optcom_mean = optcom_masked.mean(axis=-1)[:, np.newaxis]
-    optcom_std = optcom_masked.std(axis=-1)[:, np.newaxis]
+    data_optcom_masked = data_optcom[mask, :]
+    optcom_mean = data_optcom_masked.mean(axis=-1)[:, np.newaxis]
+    optcom_std = data_optcom_masked.std(axis=-1)[:, np.newaxis]
 
     # Compute temporal regression
-    optcom_z = stats.zscore(optcom_masked, axis=-1)
-    comp_pes = np.linalg.lstsq(mmix, optcom_z.T, rcond=None)[0].T  # component parameter estimates
-    resid = optcom_z - np.dot(comp_pes[:, not_ign], mmix[:, not_ign].T)
+    data_optcom_z = stats.zscore(data_optcom_masked, axis=-1)
+    # component parameter estimates
+    comp_pes = np.linalg.lstsq(mixing, data_optcom_z.T, rcond=None)[0].T
+    # Get residuals (ignored/low-variance components and unmodeled noise)
+    resid = data_optcom_z - np.dot(comp_pes[:, not_ign], mixing[:, not_ign].T)
 
     # Build time series of just BOLD-like components (i.e., MEHK) and save T1-like effect
-    mehk_ts = np.dot(comp_pes[:, acc], mmix[:, acc].T)
+    mehk_ts = np.dot(comp_pes[:, acc], mixing[:, acc].T)
     t1_map = mehk_ts.min(axis=-1)  # map of T1-like effect
     t1_map -= t1_map.mean()
     io_generator.save_file(utils.unmask(t1_map, mask), "t1 like img")
     t1_map = t1_map[:, np.newaxis]
 
     # Find the global signal based on the T1-like effect
-    glob_sig = np.linalg.lstsq(t1_map, optcom_z, rcond=None)[0]
+    gs_ts = np.linalg.lstsq(t1_map, data_optcom_z, rcond=None)[0]
```

Comment:
I wonder, does it make sense to write out this time series to the confounds file I'm adding in #1044? @handwerkerd WDYT?

Reply:
I decided to tack it onto the confounds file.